packages/zowe-explorer/src/trees/dataset/DatasetActions.ts

```diff
@@ -1526,6 +1528,20 @@ export class DatasetActions {
             await datasetProvider.checkCurrentProfile(node);
             if (Profiles.getInstance().validProfile !== Validation.ValidationType.INVALID) {
                 await vscode.workspace.fs.delete(node.resourceUri, { recursive: false });
+                if (node?.children) {
+                    node.children = node.children.filter((child) => child !== node);
+                }
+
+                const dataSetNode = node.getSessionNode() as ZoweDatasetNode;
+                const isSession = SharedContext.isSession(dataSetNode) || SharedContext.isFavoriteSearch(dataSetNode);
+                const fetchFunction = isSession
+                    ? dataSetNode.listDatasetsInRange.bind(dataSetNode)
+                    : dataSetNode.listMembersInRange.bind(dataSetNode);
+
+                const itemsPerPage = SettingsConfig.getDirectValue<number>(Constants.SETTINGS_DATASETS_PER_PAGE) ?? Constants.DEFAULT_ITEMS_PER_PAGE;
+
+                dataSetNode.paginator = new Paginator(itemsPerPage, fetchFunction);
+                dataSetNode.paginatorData = undefined;
```

Comment:
Couple things to consider for this change:

- We should only do this if `itemsPerPage` is > 0, meaning that pagination is enabled (requesting changes for this specific point)
- If I have a data set at the end of the second page of items and I delete it, this would reset the paginator to the first page. Not sure if this is a huge issue since its just cosmetic, but it could be inconvenient if a user is trying to delete multiple items on a page that isn't page 1.